src/java/org/apache/cassandra/index/sai/disk/v1/SSTableIndexWriter.java

```diff
@@ -151,15 +152,12 @@ public void complete(Stopwatch stopwatch) throws IOException
                              indexDescriptor.descriptor, FBUtilities.prettyPrintMemory(bytesAllocated), FBUtilities.prettyPrintMemory(globalBytesUsed));
             }
 
-            if (this.indexContext.isSegmentCompactionEnabled())
```

Comment:
STAR-1020 is the solution to compaction performance.

Reply:
I didn't realise that we were disabling this in the v1 code. This isn't acceptable because we are going to have a situation where we may have indexes in the field that haven't done segment compaction. If we remove all the code then these indexes aren't going to work.

Reply:
Nothing was explicitly done in this PR to make existing indexes with segments not work.  The mixed index version unit test will show if that's true.

Reply:
As was discussed during the SAI call yesterday, we shouldn't remove this expression in this PR so this hunk needs to be reverted. Instead, we can remove the `this.indexContext.isSegmentCompactionEnabled()` check in a follow-up PR that fixes the segment compaction performance issue.